File: Facebook/src/logics/adController.js
const facebookAdsService = require('../services/facebookAdsService');

const createAd = async (req, res) => {
  try {
    // Extract ad data from the request body
    const { name, adSetId, creative } = req.body;

    // Call the Facebook Ads service to create the ad
    const ad = await facebookAdsService.createAd(name, adSetId, creative);

    res.status(201).json(ad);
  } catch (error) {
    console.error('Error creating ad:', error);
    res.sendStatus(500);
  }
};

module.exports = {
  createAd,
};

File: FacebookSrv/controllers/ad_controller.py
from flask import request, jsonify
from services.facebook_ads_service import create_ad

def create_ad_controller():
    data = request.get_json()
    name = data.get('name')
    ad_set_id = data.get('ad_set_id')
    creative = data.get('creative')

    ad = create_ad(name, ad_set_id, creative)
    return jsonify(ad), 201
